mantidimaging/gui/windows/main/view.py

```diff
@@ -99,6 +106,21 @@ def show_load_dialogue(self):
         self.load_dialogue = MWLoadDialog(self)
         self.load_dialogue.show()
 
+    def load_sample_log_dialog(self):
+        stack_selector = StackSelectorDialog(main_window=self,
+                                             title="Stack Selector",
+                                             message="Which stack is the log being loaded for?")
+        stack_selector.exec()
+        stack_to_add_log_to = stack_selector.selected_stack
+
+        # Open file dialog
+        file_filter = "Log File (*.txt *.log)"
+        selected_file, _ = Qt.QFileDialog.getOpenFileName(caption="Log to be loaded",
+                                                          filter=f"{file_filter};;All (*.*)",
+                                                          initialFilter=file_filter)
+
+        self.presenter.add_log_to_sample(stack_name=stack_to_add_log_to, log_file=selected_file)
```

Comment:
This needs to handle closing via X (Close on Ubuntu), which I think returns just an empty string

Reply:
I really should have caught this! good spot!